File: 프로그래머스/0/120890. 가까운 수/가까운 수.py
def solution(array, n):
    answer = 100
    box = []
    for i in array:
        if abs(n - i) < answer:
            box = []
            answer = abs(n - i)
            box.append(i)
        elif abs(n - i) == answer:
            box.append(i)       
    return min(box)

File: 프로그래머스/0/120890. 가까운 수/가까운 수.java
import java.util.Arrays;
class Solution {
   public int solution(int[] array, int n) {
    Arrays.sort(array);
    int min_diff = Math.abs(n - array[0]);
    int answer_id = 0;
    
    for (int i = 1; i < array.length; i++) {
        int diff = Math.abs(n - array[i]);
        // 더 작은 차이가 발견되면 업데이트
        if (diff < min_diff) {
            min_diff = diff;
            answer_id = i;
        }
        // 동일한 차이면, 배열이 정렬되어 있으므로 기존 answer_id가 더 작은 값입니다.
    }
    
    return array[answer_id];
}

}